### PR TITLE
Munki staging processor

### PR DESCRIPTION
**Pull request by choules**
This pull request introduces a new shared processor which may be used to implement an automatic staging of items in a Munki repo.

To use this processor, the wycomco-recipes repo should be in AutoPKG's search path and then the processor may be referenced like this:

```
<dict>
	<key>Processor</key>
	<string>com.github.wycomco.SharedProcessors/MunkiAutoStaging</string>
</dict>
```

To enable this reference method, I have created a stub recipe as describe in the [AutoPKG wiki](https://github.com/autopkg/autopkg/wiki/Processor-Locations#shared-recipe-processors).

The processor uses common input vars from a Munki recipe, and the following new ones
* MUNKI_STAGING_CATALOG: The name of the staging catalog, defaults to "testing"
* MUNKI_PRODUCTION_CATALOG: The name of the production catalogs, defaults to "production"
* MUNKI_STAGING_DAYS: The amount of days to keep a new item in staging catalog

Currently, there is no way to force a makecatalog using the MakeCatalog.munki recipe, see this [issue](https://github.com/autopkg/recipes/issues/418) for details. So it may be the case, that even though the catalog was set correctly inside the pkginfo file, the catalogs do not get updated correctly.

### CODE REVIEW

File: SharedProcessors/MunkiAutoStaging.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+            "default": "/usr/local/munki",
+        },
+        "MUNKI_PKGINFO_FILE_EXTENSION": {
+            "description": (
+                "Extension for output pkginfo files. Default is 'plist'.",
+            ),
+            "required": False,
+            "default": "plist",
+        },
+        "MUNKI_TESTING_CATALOG": {
```
Comment (jutonium) [inline SharedProcessors/MunkiAutoStaging.py:110]: This variable should be called MUNKI_STAGING_CATALOG to make the connection to variable line 124 MUNKI_STAGING_DAYS more obvious.

File: SharedProcessors/MunkiAutoStaging.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+                if ("_metadata" not in pkginfo) or (
+                    "creation_date" not in pkginfo["_metadata"]
+                ):
+                    continue
+
+                delta = datetime.now() - pkginfo["_metadata"]["creation_date"]
+
+                if delta.days > int(self.env["MUNKI_STAGING_DAYS"]):
+                    self.output(f"Found item to promote at {file}")
+                    items_to_promote.append(file)
```
[the remark this answers is not in the document]
Comment (choules) [reply SharedProcessors/MunkiAutoStaging.py:216]: If the day is not over, it's not over. 😉
As we don't know in which intervals AutoPKG will be run, I think this is totally acceptable.

**Comment by choules** [comment]
Updated to reflect the comments above.

File: SharedProcessors/MunkiAutoStaging.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (choules) [reply SharedProcessors/MunkiAutoStaging.py:216]: As discussed, please implement MUNKI_STAGING_DAYS as float, @jutonium. Thanks!

[the remark this answers is not in the document]
Comment (jutonium) [reply SharedProcessors/MunkiAutoStaging.py]: Done. Please review.

[the remark this answers is not in the document]
Comment (jutonium) [reply SharedProcessors/MunkiAutoStaging.py]: @choules see above

**Comment by choules** [comment]
Tested locally, everything works as expected.

**Pull request merged**